LoginAndAclSetup.md

```diff
@@ -0,0 +1,178 @@
+# Setting up optional login and document level access control
+
+## Table of Contents
+
+- [Requirements](#requirements)
+- [Setting up Azure AD Apps](#setting-up-azure-ad-apps)
+  - [Manual Setup](#manual-setup)
+    - [Server App](#server-app)
+    - [Client App](#client-app)
+- [Optional Scripts](#optional-scripts)
+  - [Azure Data Lake Storage Gen2 Setup](#azure-data-lake-storage-gen2-setup)
+  - [Azure Data Lake Storage Gen2 Prep Docs](#azure-data-lake-storage-gen2-prep-docs)
+  - [Manually managing Document Level Access Control](#manually-managing-document-level-access-control)
+- [Environment Variables Reference](#environment-variables-reference)
+
+This guide demonstrates how to add an optional login and document level access control system to the sample. This system can be used to restrict access to indexed data to specific users based on what [Azure Active Directory (Azure AD) groups](https://learn.microsoft.com/azure/active-directory/fundamentals/how-to-manage-groups) they are a part of, or their [user object id](https://learn.microsoft.com/partner-center/find-ids-and-domain-names#find-the-user-object-id).
+
+![AppLoginArchitecture](./docs/applogincomponents.png)
+
+## Requirements
+
+**IMPORTANT:** In order to add optional login and document level access control, you'll need the following in addition to the normal sample requirements
+
+* **Azure account permissions**: Your Azure account must have [permission to manage applications in Azure AD](https://learn.microsoft.com/azure/active-directory/develop/quickstart-register-app#prerequisites).
```

Comment:
Would it be better to directly link to https://learn.microsoft.com/en-us/azure/active-directory/roles/permissions-reference#cloud-application-administrator ? That's what we do for the other roles in the main README